src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Debug/LaunchSettings.cs

```diff
@@ -37,6 +39,34 @@ public LaunchSettings(LaunchSettingsData settingsData, string activeProfile = nu
             _activeProfileName = activeProfile;
         }
 
+        public LaunchSettings(IWritableLaunchSettings settings)
+        {
+            Profiles = ImmutableList<ILaunchProfile>.Empty;
+            foreach(var profile in settings.Profiles)
+            {
+                Profiles = Profiles.Add(new LaunchProfile(profile));
+            }
+            
+            // For global settings we want to make new copies of each entry so that the snapshot remains immutable. If the object implements 
+            // ICloneable that is used, otherwise, it is serialized back to json, and a new object rehydrated from that
+            GlobalSettings = ImmutableDictionary<string, object>.Empty;
+            foreach(var kvp in settings.GlobalSettings)
+            {
+                if(kvp.Value is ICloneable)
+                {
+                    GlobalSettings = GlobalSettings.Add(kvp.Key, ((ICloneable)kvp.Value).Clone());
+                }
+                else
+                {
+                    string jsonString = JsonConvert.SerializeObject(kvp.Value, Formatting.Indented, new JsonSerializerSettings() { NullValueHandling = NullValueHandling.Ignore});
```

Comment:
I couldn't come up with a better one that would do a proper deep copy. The formatting matters because it will create an empty value which may become non-null in the clone. When it is later persisted, the empty entry will be created in the launch settings file.